src/codecs/tiff.rs

```diff
@@ -74,12 +76,29 @@ where
             }
         };
 
+        let original_color_type = match tiff_color_type {
+            tiff::ColorType::CMYK(8) => ExtendedColorType::Cmyk8,
+            _ => color_type.into(),
+        };
+
         Ok(TiffDecoder {
             dimensions,
             color_type,
+            original_color_type,
             inner: Some(inner),
         })
     }
+
+    fn total_buffer_bytes(&self) -> u64 {
+        let dimensions = self.dimensions();
+        let total_pixels = u64::from(dimensions.0) * u64::from(dimensions.1);
+        let bytes_per_pixel = if self.original_color_type == ExtendedColorType::Cmyk8 {
```

Comment:
That needs a `bytes_per_pixel` for `ExtendedColorType`

Reply:
We could have a `ExtendedColorType::bits_per_pixel` method, but thanks to variants like `ExtendedColorType::L1` having an integers bytes per pixel isn't going to work :)